node/src/main/java/io/airlift/node/NodeInfo.java

```diff
@@ -58,6 +61,7 @@ public class NodeInfo
     private final InetAddress bindIp;
     private final long startTime = System.currentTimeMillis();
     private final Map<String, String> annotations;
+    private static boolean preferIpv6ForTest;
```

Comment:
This approach to testing is problematic.
1. It introduces non-thread-safe state in a class that is meant to be thread-safe
2. It adds mutable global state, which is harder to reason about.
3. It add testing code in the main execution path, which adds complexity and potential for things breaking.

A better way of modeling it would be to pass the preference in the constructor of this object (e.g., via the config object)

Reply:
Make sense. Thank you for feedback. Fixed it